src/analysis/convex.lean

```diff
@@ -748,6 +748,10 @@ calc
     add_le_add (mul_le_mul_of_nonneg_left (le_max_left _ _) ha) (mul_le_mul_of_nonneg_left (le_max_right _ _) hb)
   ... ≤ max (f x) (f y) : by rw [←add_mul, hab, one_mul]
 
+/-- This instance is necessary to guide class instance search in the lemma below. -/
```

Comment:
In this case, I wouldn't use a docstring, but an ordinary comment. After all, this text is only meaningful in the context of this file. In a tooltip it doesn't tell the user anything useful.

Reply:
Ok, done.

Reply:
This should be a local instance if it's only needed for this lemma. It could have performance effects on the rest of the library otherwise (we're having some instance depth problems, as I'm sure you've seen)

Reply:
Writing `local instance` gives me `invalid local/reserve notation, 'infix', 'infixl', 'infixr', 'prefix', 'postfix' or 'notation' expected`. How can I make it a local instance?

Reply:
Write `def some_name` instead of `instance`.
Then afterward write `local attribute [instance] some_name`.